flocker/control/test/test_persistence.py

```diff
@@ -761,11 +768,51 @@ def test_upgrade_configuration_versions(self, versions):
         """
         source_version, target_version = versions
         configs_dir = FilePath(__file__).sibling('configurations')
-        source_json_file = b"configuration_v%d.json" % source_version
-        target_json_file = b"configuration_v%d.json" % versions[1]
+
+        # Choose the latest configuration number available for the given
+        # version of the config. The configuration has increased in complexity
+        # over time, so we have added additional configurations to verify that
+        # the new fields can be correctly upgraded.
+        source_json_glob = b"configuration_*_v%d.json" % source_version
+        source_jsons = sorted(configs_dir.globChildren(source_json_glob),
+                              key=lambda x: x.basename())
+        config_num = int(source_jsons[-1].basename().split('_')[1])
+
+        source_json_file = b"configuration_%d_v%d.json" % (config_num,
+                                                           versions[0])
+        target_json_file = b"configuration_%d_v%d.json" % (config_num,
+                                                           versions[1])
         source_json = configs_dir.child(source_json_file).getContent()
         target_json = configs_dir.child(target_json_file).getContent()
         upgraded_json = migrate_configuration(
             source_version, target_version,
             source_json, ConfigurationMigration)
         self.assertEqual(json.loads(upgraded_json), json.loads(target_json))
+
+
+class LatestGoldenFilesValid(TestCase):
+    """
+    Tests for the latest golden files to ensure they have not regressed.
+    """
+
+    def test_can_create_latest_golden(self):
+        """
+        The latest golden files should be identical to ones generated from
+        HEAD.
+        """
+        configs_dir = FilePath(__file__).sibling('configurations')
+        for i, deployment in enumerate(TEST_DEPLOYMENTS, start=1):
+            encoding = wire_encode(
+                Configuration(version=_CONFIG_VERSION, deployment=deployment)
+            )
+            path = configs_dir.child(
+                b"configuration_%d_v%d.json" % (i, _CONFIG_VERSION)
+            )
+            self.assertEqual(
+                encoding, path.getContent(),
+                "Golden test file %s can not be generated from HEAD. Please "
+                "review the python files in that directory to re-generate "
+                "that file if you have intentionally changed the backing test "
+                "data. You might need to rev the model if you are "
+                "intentionally changing the model." % path.path
+            )
```

Comment:
Looks good. I tried running this test against master, but it failed for not finding the new second configuration file.

```
(4413-master) (master ✓? =)[~/.../HybridLogic/flocker]$ trial flocker.control.test.test_persistence.LatestGoldenFilesValid
flocker.control.test
  test_persistence
    LatestGoldenFilesValid ...                                          [ERROR]

===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/.virtualenvs/4413-master/lib/python2.7/site-packages/twisted/trial/runner.py", line 605, in loadByNames
    things.append(self.findByName(name))
  File "/home/richard/.virtualenvs/4413-master/lib/python2.7/site-packages/twisted/trial/runner.py", line 409, in findByName
    return reflect.namedAny(name)
  File "/home/richard/.virtualenvs/4413-master/lib/python2.7/site-packages/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/home/richard/.virtualenvs/4413-master/lib/python2.7/site-packages/twisted/python/reflect.py", line 242, in _importAndCheckStack
    return __import__(importName)
  File "/home/richard/projects/HybridLogic/flocker/flocker/control/test/test_persistence.py", line 79, in <module>
    'configurations').child(b"configuration_1_v1.json").getContent()
  File "/home/richard/.virtualenvs/4413-master/lib/python2.7/site-packages/twisted/python/filepath.py", line 294, in getContent
    fp = self.open()
  File "/home/richard/.virtualenvs/4413-master/lib/python2.7/site-packages/twisted/python/filepath.py", line 1010, in open
    return open(self.path, mode + 'b')
exceptions.IOError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/flocker/control/test/configurations/configuration_1_v1.json'

flocker.control.test.test_persistence.LatestGoldenFilesValid
-------------------------------------------------------------------------------
Ran 1 tests in 0.001s

FAILED (errors=1)

```